### ISSUE TITLE
Make the applications in FreeBSD accessible

### ISSUE THREAD

**Issue by probonopd** [helloSystem/hello#173]
We need a way to make the applications in FreeBSD accessible.

For example, by creating one "placeholder `.app` bundle" for each application in FreeBSD.
Like we currently do with LibreOffice and the like.

For this we need:

* The application name in correct spelling (e.g., "Qalculate!", "KTurtle", "SpeedCrunch"). Not the all-lowercase atrocity that package managers typically use. We can get this from
  * `pkg-descr` *if* the second word is "is", which is often the case but not always
  * The AppStream metainfo file at , `/usr/local/share/metainfo/*, in the "name" key in the "component"  element, *if* the package has such a file (but for this we need to have access to the package)
  * The `*.desktop` file at `/usr/local/share/applications/`, in the `Name=` key in the `[Desktop Entry]` element (but for this we need to have access to the package)
* The application icon as an 128x128 pixels png from `/usr/local/share/icons/.*/.*/apps/.*[png|svg]`. Unfortunately some packages only ship scalable versions so we need to take the svg and convert it in those cases (but for this we need to have access to the package)
* The absolute path of the (main) executable for the application.
 
Packages that contain no `*.desktop` file at `/usr/local/share/applications/` are likely not GUI applications and are hence not of interest to us. An exception may be GNUstep related application. (It's a shame that we have to rely on the pesky Desktop Entry Specification to tell an application package from a non-application package.)

Regarding "we need to have access to the package":
Having to download 100 GB worth of packages is wasteful, slow, and expensive.

Possibly we can use a combination of https://github.com/jbd/httpfs2 or http://httpfs.sourceforge.net/ and libarchive to get packages "mounted" without downloading the entirety of them.

Or possibly we can get https://www.freshports.org/ interested enough to provide this data.

Note to self: FreeBSDPortsToApps.ipnynb

**Comment by probonopd** [comment]
Looks like https://github.com/ghostbsd/software-station does also not have access to this kind of information.

**Comment by linuxkettle** [comment]
Do you want a script of some sort to convert every package in the FreeBSD binary repository into an app bundle? And then store all of that on the user's system? Wouldn't that fill the Applications folder with apps somebody doesn't really need or want? Or would we have a separate folder that you download and it's full of placeholders like those in /usr/share/ports that download the meat and the vegetables when you need them to.

I'm in favour of the app shop idea. This will get messy.

**Comment by probonopd** [comment]
> Do you want a script of some sort to convert every package in the FreeBSD binary repository into an app bundle? And then store all of that on the user's system?

Exactly. Not "every binary package" but the "most popular GUI applications". And only as a placeholder application bundle that works like the one we currently ship for e.g., LibreOffice. It takes up almost no space before the user actually runs it for the first time, at which point the application is downloaded.
